extensions/wikia/FeedsAndPosts/JustUpdatedAPIProxy.class.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+namespace Wikia\FeedsAndPosts;
+
+class JustUpdatedAPIProxy {
+
+	const LIMIT = 4;
+	const MINOR_CHANGE_THRESHOLD = 100;
+
+	const IMAGE_WIDTH = 150;
+	const IMAGE_RATIO = 3/4;
+
+	private function requestAPI( $params ) {
+		$api = new \ApiMain( new \FauxRequest( $params ) );
+		$api->execute();
+
+		return $api->GetResultData();
+	}
+
+	private function getThumbnailUrl( $url ) {
+		try {
+			return \VignetteRequest::fromUrl( $url )
+				->zoomCrop()
+				->width( self::IMAGE_WIDTH )
+				->height( floor( self::IMAGE_WIDTH / self::IMAGE_RATIO ) )
+				->url();
+		}
+		catch ( \Exception $exception ) {
+			\Wikia\Logger\WikiaLogger::instance()
+				->warning( "Invalid thumbnail url provided for recent updates module",
+					[
+						'thumbnailUrl' => $url,
+						'message' => $exception->getMessage(),
+					] );
+
+			return $url;
+		}
+	}
+
+	private function filterOutSmallChangesAndCleanUp( $articles ) {
+		$articlesWithMinorChange = [];
+		$articlesWithMajorChange = [];
+		$mainPageId = \Title::newMainPage()->getArticleID();
+
+		foreach ( $articles as &$article ) {
+			$diffSize = abs( $article['newlen'] - $article['oldlen'] );
+			$articleId = $article['pageid'];
+
+			unset(
+				$article['newlen'],
```

Comment:
instead of this `unset` can we just create a new object with only the data we need? IMO it will be much more readable

Reply:
Totally make sense :)